### PR TITLE
added support for date-time format

### PR DESCRIPTION
**Pull request by ilaif**
Closes https://github.com/fastify/fast-json-stringify/issues/194

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

#### Additional notes
- As this is the first time I'm touching this beautiful piece of code, please criticize the way I implemented the feature and that I didn't miss any important cases.
- I only implemented the `date-time` format, but `date` and `time` also exist. I think this is a rarer use-case that might require more complex code to achieve safely. 

Thanks!

### CODE REVIEW

**Comment by ilaif** [comment]
Done.

**Comment by Eomm** [comment]
@ilaif do you think it is possible to support the format `date` and `time` also?

**Comment by ilaif** [comment]
Yeah, sure, will have a look.

**Comment by mcollina** [comment]
What's the status @ilaif?
@Eomm do you think this is landable?

**Comment by jsumners** [comment]
@delvedor it is part of the spec -- https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.7.3

**Comment by Eomm** [comment]
@ilaif would you mind to carry this PR to the finish line? :muscle:

File: package.json [hunk 1]
```diff
@@ -35,11 +35,13 @@
     "is-my-json-valid": "^2.20.0",
     "json-strify": "^0.1.7",
     "long": "^4.0.0",
+    "moment": "^2.24.0",
     "pre-commit": "^1.2.2",
     "proxyquire": "^2.1.3",
     "semver": "^7.1.0",
     "standard": "^14.0.0",
     "tap": "^12.6.5",
+    "tap-mocha-reporter": "^3.0.9",
```
Comment (mcollina) [inline package.json:44]: Why was this added?

Comment (ilaif) [reply package.json]: My bad, missed it while merging from upstream.

**Comment by ilaif** [comment]
@mcollina @Eomm Hey guys,

I've added support for `date` and `time` formats as defined by `is-my-json-valid` that determines the following formats:
- `date`: `YYYY-MM-DD`
- `time`: `HH:mm:ss` (without UTC offset)

Another important addition is adding moment.js as a runtime dependency of the project so I can use it to check if an object is an instance of moment, and to format to `date` and `time` formats more easily. Let me know if this sounds reasonable to you.

File: index.js [hunk 2]
```diff
@@ -248,6 +282,36 @@ function $asBooleanNullable (bool) {
   return bool === null ? null : $asBoolean(bool)
 }
 
+function $asDatetime (date) {
+  if (date instanceof Date) {
+    return '"' + date.toISOString() + '"'
+  } else if (date instanceof moment) {
```
Comment (mcollina) [inline index.js:288]: Unfortunately this check will not work because a user can have more than one version of moment in their dep tree.

File: index.js [hunk 3]
```diff
@@ -248,6 +282,36 @@ function $asBooleanNullable (bool) {
...
+    return '"' + date.toISOString() + '"'
+  } else {
+    return $asString(date)
+  }
+}
+
+function $asDate (date) {
+  if (date instanceof Date) {
+    return '"' + moment(date).format('YYYY-MM-DD') + '"'
+  } else if (date instanceof moment) {
```
Comment (mcollina) [inline index.js:298]: Unfortunately this check will not work because a user can have more than one version of moment in their dep tree.

File: index.js [hunk 4]
```diff
@@ -248,6 +282,36 @@ function $asBooleanNullable (bool) {
...
+    return '"' + date.format('YYYY-MM-DD') + '"'
+  } else {
+    return $asString(date)
+  }
+}
+
+function $asTime (date) {
+  if (date instanceof Date) {
+    return '"' + moment(date).format('HH:mm:ss') + '"'
+  } else if (date instanceof moment) {
```
Comment (mcollina) [inline index.js:308]: Unfortunately this check will not work because a user can have more than one version of moment in their dep tree.

**Comment by mcollina** [comment]
I'm also -1 in adding `moment` as a runtime dependency.

**Comment by ilaif** [comment]
Fair enough, will write a custom formatting function and use duck typing.

**Comment by Eomm** [comment]
@mcollina I don't have permissions to release this module, could you publish a new version?

**Comment by mcollina** [comment]
Done in 1.19.0
